src/main/scala/bookbot/BookBot.scala

```diff
@@ -5,11 +5,13 @@ import io.circe._, io.circe.parser._
 import bookbot.models.errors.{Error, HttpError, TransformError}
 import Error.ErrorOr
 import bookbot.models.{GoogleResponse, Volume, Book}
+import bookbot.models.TitleAndAuthor
+import TitleAndAuthor._
+import cats.syntax.either._
+import cats.implicits._
 
 object BookBot {
-  type TitleAndAuthor = (String, String)
-
-  def findLinkFrom = getTitleAndAuthor _ andThen getBook andThen getLink
+  def findLinkFrom(searchString: String) = TitleAndAuthor.from(searchString).flatMap(getBook andThen getLink)
```

Comment:
Functional Composition is a good indicator of FP techniques, here in Scala as `andThen`.